spring-core/src/main/java/org/springframework/core/annotation/AnnotatedElementUtils.java

```diff
@@ -1243,23 +1297,32 @@ else if (element instanceof Class) {
 
 		for (Class<?> iface : ifcs) {
 			if (AnnotationUtils.isInterfaceWithAnnotatedMethods(iface)) {
-				try {
-					Method equivalentMethod = iface.getMethod(method.getName(), method.getParameterTypes());
+				Method equivalentMethod = findEquivalentMethod(iface, method);
+				if (equivalentMethod != null) {
 					T result = searchWithFindSemantics(equivalentMethod, annotationType, annotationName, containerType,
 							processor, visited, metaDepth);
 					if (result != null) {
 						return result;
 					}
 				}
-				catch (NoSuchMethodException ex) {
-					// Skip this interface - it doesn't have the method...
-				}
 			}
 		}
 
 		return null;
 	}
 
+	private static Method findEquivalentMethod(Class<?> clazz, Method method) {
+		Method[] candidates = clazz.getDeclaredMethods();
+		for (Method candidate : candidates) {
+			if (candidate.getName().equals(method.getName()) && Arrays.equals(
+					candidate.getParameterTypes(), method.getParameterTypes())) {
```

Comment:
May I suggest a pre-check `candidate.getParameterCount() == method.getParameterCount())` in order to avoid possible allocations coming from `getParameterTypes()`?

Reply:
Good suggestions. I've updated the PR. Thanks!